ballerina/types.bal

```diff
@@ -13,50 +13,104 @@
+# + password - The password for the Redis database  
+# + options - Other connection options of the connection configuration
+@display {label: "Connection Parameters"}
+type ConnectionParams record {|
     @display {label: "Host"}
-    string host?;
+    string host = "localhost";
+    @display {label: "Port"}
+    int port = 6379;
+    @display {label: "Username"}
+    string username?;
     @display {label: "Password"}
     string password?;
     @display {label: "Connection Options"}
     Options options = {};
 |};
 
+# The redis Connection URI based configurations. This can become useful when working with 
+# managed Redis databases, where the cloud provider usually provides a connection URI.
+#
+# + uri - The connection URI for the Redis database
+@display {label: "Connection URI"}
+type ConnectionUri string;
+
 # Connection options for Redis client endpoint.
 #
-# + clientName - The clientName of the connection
-# + connectionPooling - Boolean value depending on whether the connection
-# pooling is enabled or not
-# + isClusterConnection - Whether to enable cluster connection or not
-# + ssl - Boolean value depending on whether SSL is enabled or not
-# + startTls - Boolean value depending on whether startTLS is enabled or not
-# + verifyPeer - Boolean value depending on whether peer verification is
-# enabled or not
-# + database - The database to be used with the connection
-# + connectionTimeout - The timeout value for the connection
+# + clientName - Name of the client
+# + database - Database index which the client should interact with. Not applicable for cluster connections
+# + connectionTimeout - Connection timeout in seconds
+@display {label: "Connection Options"}
 public type Options record {|
     @display {label: "Client Name"}
     string clientName?;
-    @display {label: "Connection Pooling Enabled"}
-    boolean connectionPooling = false;
-    @display {label: "Cluster Connection Enabled"}
-    boolean isClusterConnection = false;
-    @display {label: "SSL Enabled"}
-    boolean ssl = false;
-    @display {label: "STARTTLS Enabled"}
-    boolean startTls = false;
-    @display {label: "Peer Verification Enabled"}
-    boolean verifyPeer = false;
-    @display {label: "Database"}
-    int database = -1;
+    @display {label: "Database Index"}
+    int database = 0;
     @display {label: "Connection Timeout"}
-    int connectionTimeout = -1;
+    int connectionTimeout = 60;
+|};
+
+# Configurations for secure communication with the Redis server.
+#
+# + cert - Configurations associated with `crypto:TrustStore` or single certificate file that the client trusts  
+# + key - Configurations associated with `crypto:KeyStore` or combination of certificate and private key of the client  
+# + protocols - List of protocols used for the connection established to Redis Server, such as TLSv1.2, TLSv1.1, TLSv1.
+# + ciphers - List of ciphers to be used for SSL connections
+# + verifyPeer - Whether peer verification is enabled
```

Comment:
Is this related to mTLS ? If so can't we align it with other connectors or library modules ?

Reply:
This is not related to mTLS. This option decides how the verification should be done (no verifications, only cert, cert + hostname). 
I've added some improvements related to this option with https://github.com/ballerina-platform/module-ballerinax-redis/pull/218/commits/73f278f552f848180bcc2f4ac6d1c197a86e05e8. Please check and verify.